slatedb/src/sst_iter.rs

```diff
@@ -401,6 +595,7 @@ impl KeyValueIterator for SstIterator<'_> {
             match next_entry {
                 Some(kv) => {
                     if self.view.contains(&kv.key) {
+                        self.maybe_mark_key_found(kv.key.as_ref());
```

Comment:
if you take my suggestion above it might be better to just delegate the stats-recording to the filter evaluator. So it can have fns like `notify_key_found(&self, key: &[u8])` and `notify_finished_iteration` and it can just record stats there

Reply:
I like this, will make the change!

Reply:
OK, it's ready for another round of review. I updated it to use a wrapping iterator - ultimately it was the cleanest